susemanager-utils/susemanager-sls/src/modules/kiwi_info.py

```diff
@@ -69,25 +69,37 @@ def guess_buildinfo(dest):
     return ret
 
 # Kiwi NG
+_kiwi_result_script = """
+import sys
+import pickle
+import json
+ret = {}
+with open(sys.argv[1], 'rb') as f:
+    result = pickle.load(f)
+    ret['arch'] = result.xml_state.host_architecture
+    ret['basename'] = result.xml_state.xml_data.name
+    ret['type'] = result.xml_state.build_type.image
+    ret['filesystem'] = result.xml_state.build_type.filesystem
+    ret['initrd_system'] = result.xml_state.build_type.initrd_system
+    print(json.dumps(ret))
+"""
+
 def parse_kiwi_result(dest):
     path = os.path.join(dest, 'kiwi.result')
     ret = {}
     if __salt__['file.file_exists'](path):
-        try:
-            # pickle depends on availability of python kiwi modules
-            # which are not under our control so there is certain risk of failure
-            # return empty dict in such case
-            # the caller should handle all values as optional
-            with open(path, 'rb') as f:
-                result = pickle.load(f)
-                ret['arch'] = result.xml_state.host_architecture
-                ret['basename'] = result.xml_state.xml_data.name
-                ret['type'] = result.xml_state.build_type.image
-                ret['filesystem'] = result.xml_state.build_type.filesystem
-                ret['initrd_system'] = result.xml_state.build_type.initrd_system
-        except:
-            log.exception("Loading kiwi.result")
-            # continue with empty dict
+        # pickle depends on availability of python kiwi modules
+        # which are not under our control so there is certain risk of failure
+        # also, the kiwi libraries may not be available in salt bundle
+        # -> parse the file via wrapper script using system python3
+        #
+        # return empty dict on failure
+        # the caller should handle all values as optional
+        result = __salt__['cmd.exec_code_all']('/usr/bin/python3', _kiwi_result_script, args=[path])
+        if result['retcode'] == 0:
+            ret = json.loads(result['stdout'])
+        # else return empty dict
```

Comment:
Having this line commented is intended?

Reply:
a bit redundant, on my taste, but makes this part more descriptive

Reply:
yes, it is intended